src/tests/ftest/launch.py

```diff
@@ -1459,886 +1108,1972 @@ def ignore_unknown(self, node):  # pylint: disable=no-self-use,unused-argument
+            test (TestInfo): the test information
+            tag_filters (list): optional '--filter-by-tags' arguments
+            sparse (bool): whether or not to provide sparse output of the test execution
+            failfast (bool): _description_
+            extra_yaml (list): additional yaml files to include on the command line
+
+        Returns:
+            list: avocado run command
+
+        """
+        command = ["avocado"]
+        if not sparse and self.major >= 82:
+            command.append("--show=test")
+        command.append("run")
+        if self.major >= 82:
+            command.append("--ignore-missing-references")
+        else:
+            command.extend(["--ignore-missing-references", "on"])
+        if self.major >= 83:
+            command.append("--disable-tap-job-result")
+        else:
+            command.extend(["--html-job-result", "on"])
+            command.extend(["--tap-job-result", "off"])
+        if not sparse and self.major < 82:
+            command.append("--show-job-log")
+        if tag_filters:
+            command.extend(tag_filters)
+        if failfast:
+            command.extend(["--failfast", "on"])
+        command.extend(["--mux-yaml", test.yaml_file])
+        if extra_yaml:
+            command.extend(extra_yaml)
+        command.extend(["--", str(test)])
+        return command
+
+
+class RemoteCommandResult():
+    """Stores the command result from a Task object."""
+
+    class ResultData():
+        # pylint: disable=too-few-public-methods
+        """Command result data for the set of hosts."""
+
+        def __init__(self, command, returncode, hosts, stdout, timeout):
+            """Initialize a ResultData object.
+
+            Args:
+                command (str): the executed command
+                returncode (int): the return code of the executed command
+                hosts (NodeSet): the host(s) on which the executed command yielded this result
+                stdout (list): the result of the executed command split by newlines
+                timeout (bool): indicator for a command timeout
+            """
+            self.command = command
+            self.returncode = returncode
+            self.hosts = hosts
+            self.stdout = stdout
+            self.timeout = timeout
+
+    def __init__(self, command, task):
+        """Create a RemoteCommandResult object.
+
+        Args:
+            command (str): command executed
+            task (Task): object containing the results from an executed clush command
+        """
+        self.output = []
+        self._process_task(task, command)
+
+    @property
+    def passed(self):
+        """Did the command pass on all the hosts.
+
+        Returns:
+            bool: if the command was successful on each host
+
+        """
+        timeout = all(data.timeout for data in self.output)
+        non_zero = any(data.returncode != 0 for data in self.output)
+        return not non_zero and not timeout
```

Comment:
So this would return True if 1, but not all, hosts timed out?

Reply:
timeout should use `any()` not `all()`.  Fixed, thanks.